addons/web_linkedin/models/web_linkedin.py

```diff
@@ -0,0 +1,428 @@
+        result = ""
+        status = ""
+        try:
+            if type.upper() == "GET":
+                data = werkzeug.url_encode(params)
+                req = urllib2.Request(pre_uri + uri + "?" + data)
+                for header_key, header_val in headers.iteritems():
+                    req.add_header(header_key, header_val)
+            elif type.upper() == 'POST':
+                req = urllib2.Request(pre_uri + uri, params, headers)
+            else:
+                raise ('Method not supported [%s] not in [GET, POST]!' % (type))
+            request = urllib2.urlopen(req)
+            status = request.getcode()
+            if int(status) in (204, 404):  # Page not found, no response
+                result = {}
+            else:
+                content = request.read()
+                result = simplejson.loads(content)
+        except urllib2.HTTPError, e:
+            #Should simply raise exception or simply add logger
+            if e.code in (400, 410):
+                raise e
+
+            if e.code == 401:
+                raise except_auth('AuthorizationError', {'url': self._get_authorize_uri(from_url=self.env.context.get('from_url'), scope=self.env.context.get('scope'))})
+            #TODO: Should handle 403 for throttle limit and should display user freindly message
+            status = e.code
+            _logger.exception("Bad linkedin request : %s !" % e.read())
+        except urllib2.URLError, e:
+            _logger.exception("Either there is no connection or remote server is down !")
+        return (status, result)
+
+    def _get_authorize_uri(self, from_url, scope=False):
+        """ This method return the url needed to allow this instance of OpenErp to access linkedin application """
+        state_obj = dict(d=scope, f=from_url)
+        base_url = self.get_param_parameter('web.base.url')
+        client_id = self.get_param_parameter('web.linkedin.apikey')
+
+        params = {
+            'response_type': 'code',
+            'client_id': client_id,
+            'state': simplejson.dumps(state_obj),
+            'redirect_uri': base_url + '/linkedin/authentication',
+        }
+
+        uri = self.get_uri_oauth(a='authorization') + "?%s" % werkzeug.url_encode(params)
+        return uri
+
+    def set_all_tokens(self, token_datas):
+        data = {
+            'linkedin_token': token_datas.get('access_token'),
+            'linkedin_token_validity': datetime.now() + timedelta(seconds=token_datas.get('expires_in'))
+        }
+        self.env.user.sudo().write(data)
+
+    def need_authorization(self):
+        if not self.env.user.linkedin_token_validity or \
+                datetime.strptime(self.env.user.linkedin_token_validity.split('.')[0], DEFAULT_SERVER_DATETIME_FORMAT) < (datetime.now() + timedelta(minutes=1)):
+            return True
+        return False
+
+    def get_param_parameter(self, parameter):
+        return self.env['ir.config_parameter'].sudo().get_param(parameter)
```

Comment:
Oh please don't do that. You just created a public method that allow to access any config parameter. :cry: